Update/cart slice create async thunk utilization

Ahmed-Fahmy-Dev opened the pull request:
utilize createAsyncThunk toolkit function in cart Slice

Ahmed-Fahmy-Dev committed:
create: utility function
- a pure function that writes to the local storage

Ahmed-Fahmy-Dev committed:
fix: function argument name
- argument name changed to match product itemNo property

Ahmed-Fahmy-Dev committed:
update: esLint config updated
- rule disabled to be able to write ternary statements in functions

Ahmed-Fahmy-Dev committed:
update: cart slice logic updated
- redux toolkit createAsyncThunk utilized

Ahmed-Fahmy-Dev committed:
update: cart slice actions dispatching
- dispatched actions updated according to the new thunk logic
within the corresponding components.

Ahmed-Fahmy-Dev committed:
explanatory comments and JSDocs added to cartSlice

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into update/cartSlice-createAsyncThunk-utilization

HIMDanny commented on client/src/store/slices/cartSlice/cartSlice.js:
```diff
@@ -1,158 +1,248 @@
+
+/**
+ * deletes online cart from DB
+ * @returns nothing
```
I suggest using void instead of nothing or completely delete this return comment
```suggestion
 * @returns void
```

Ahmed-Fahmy-Dev committed:
hotfix: cart slice
- deleted test paylod when dispatching fetchCart
- deleted uneeded setIsLoading reducer
